src/panels/config/automation/ha-automation-editor.ts

```diff
@@ -501,13 +503,64 @@ export class HaAutomationEditor extends KeyboardShortcutMixin(LitElement) {
         )
         .then(
           (config) => {
-            // Normalize data: ensure trigger, action and condition are lists
-            // Happens when people copy paste their automations into the config
+            // Normalize data: Ensure trigger, action and condition are lists.
+            // Happens when people copy paste their automations into the config.
             for (const key of ["trigger", "condition", "action"]) {
               const value = config[key];
               if (value && !Array.isArray(value)) {
                 config[key] = [value];
               }
+
+              // Ensure the times of a trigger are a list. Then condense them into a single
+              // trigger first (in case the user has multiple time triggers), then split them into
+              // two triggers: One for literal time values and one for input_datetime entities.
+              // Needed since for each of those types, we will need a separate trigger row.
+              if (key === "trigger") {
+                const literalTimeTrigger: TimeTrigger = {
+                  platform: "time",
+                  at: [],
+                };
+                const entityTimeTriger: TimeTrigger = {
+                  platform: "time",
+                  at: [],
+                };
+
+                for (const trigger of config[key]?.filter(
+                  (x) => x.platform === "time"
+                ) || []) {
+                  const timeTrigger = trigger as TimeTrigger;
+                  if (!Array.isArray(timeTrigger.at)) {
+                    timeTrigger.at = [timeTrigger.at];
+                  }
+
+                  for (const at of timeTrigger.at) {
+                    if (at.startsWith("input_datetime.")) {
+                      literalTimeTrigger.at.push(at);
+                    } else {
+                      entityTimeTriger.at.push(at);
+                    }
+                  }
+                }
+
+                // Remove old time triggers and replace with the two new collected ones
+                config[key] = config[key]?.filter((x) => x.platform !== "time");
+                if (entityTimeTriger.at.length)
+                  config[key].push(entityTimeTriger);
+                if (literalTimeTrigger.at.length)
+                  config[key].push(literalTimeTrigger);
+              }
+
+              // Ensure the states of a condition are a list.
+              if (key === "condition") {
+                for (const cond of config[key]?.filter(
+                  (x) => x.condition === "state"
+                ) || []) {
+                  const stateCond = cond as StateCondition;
+                  if (!Array.isArray(stateCond.state)) {
+                    stateCond.state = [stateCond.state];
+                  }
+                }
+              }
```

Comment:
We should not do this. The specific editors should deal with the different possible configs.